tpm/constants.go

```diff
@@ -71,6 +72,62 @@ const (
 	tpmCapFlagPermanent    uint32 = 0x00000108
 )
 
+// permission type
+type permission uint32
+
+// NV Permissions and Operations
+// Note: Permissions are summable
+const (
+	nvPerPPWrite    permission = 0x00000001
+	nvPerOwnerWrite permission = 0x00000002
+	nvPerAuthWrite  permission = 0x00000004
+	nvPerWriteAll   permission = 0x00000800
+	// Warning: The Value 0x00001000 is
+	// defined in the spec as
+	// TPM_NV_PER_WRITEDEFINE, but it is
+	// not included directly in this
+	// code because it locks the given
+	// NV Index permanently if used
+	// incorrectly. This operation can't
+	// be undone in any way. Do not use
+	// this value unless you know what
+	// you're doing!
+	nvPerWriteSTClear permission = 0x00002000
+	nvPerGlobalLock   permission = 0x00004000
+	nvPerPPRead       permission = 0x00008000
+	nvPerOwnerRead    permission = 0x00100000
+	nvPerAuthRead     permission = 0x00200000
+	nvPerReadSTClear  permission = 0x80000000
+)
+
+// PermMap : Map of TPM_NV_Permissions to its strings for convenience
+var PermMap = map[permission]string{
```

Comment:
Make this lower-case so its not available as the package API. Users of our package should be able to get a textual representation of the permission by calling`String()` on the `Permission` object.